src/_pytest/pytester.py

```diff
@@ -534,13 +535,13 @@ class Testdir:
     class TimeoutExpired(Exception):
         pass
 
-    def __init__(self, request, tmpdir_factory):
+    def __init__(self, request: FixtureRequest, tmpdir_factory: TempdirFactory) -> None:
         self.request = request
-        self._mod_collections = WeakKeyDictionary()
+        self._mod_collections = WeakKeyDictionary()  # type: ignore[var-annotated] # noqa: F821
         name = request.function.__name__
         self.tmpdir = tmpdir_factory.mktemp(name, numbered=True)
         self.test_tmproot = tmpdir_factory.mktemp("tmp-" + name, numbered=True)
-        self.plugins = []
+        self.plugins = []  # type: ignore[var-annotated] # noqa: F821
```

Comment:
Mypy's [`allow_untyped_globals `](https://mypy.readthedocs.io/en/stable/config_file.html#miscellaneous-strictness-flags) option might be cleaner than adding these comments to each case?

Reply:
@Zac-HD yeah.  Usually we're typing them then also, but e.g. here I'd rather get a bit more typing than nothing.

@bluetech what do you think about using the option?

Reply:
I generally wouldn't want to turn off strictness flags, only add more :)